### PR TITLE
Invalidate GITHUB_TOKEN on 401

### PR DESCRIPTION
**Pull request by Frassle**
Part one of fixing https://github.com/pulumi/pulumi/issues/17217. This disables and retries the plugin http request without GITUHB_TOKEN if we get a 401 error.

This should help with cases where people have tokens that don't have access to public resources. We'll see the 401 and retry without the token set.

This doesn't fix the other part of #17217 to retry 403 error without the token.

### CODE REVIEW

**Comment by tgummerer** [comment]
https://github.com/pulumi/pulumi/pull/17351/files#diff-196f5b7811e033d2523373f8138cca8c54eba850b419591283cd88a35828ac12R471-R473

This is outside the diff, so I can't comment on it. But this is now not always technically correct, and might confuse users if they have a token set, but still get this error message because we disabled the token.

**Comment by Frassle** [comment]
Yeh maybe we need to track we disabled and tell them to use a different GITHUB_TOKEN

**Review by tgummerer — APPROVED** [review]

File: sdk/go/common/workspace/plugins.go [hunk 1]
```diff
@@ -460,7 +473,11 @@ func (source *githubSource) getHTTPResponse(
 
 	addAuth := ""
 	if source.token == "" {
-		addAuth = " You can set GITHUB_TOKEN to make an authenticated request with a higher rate limit."
+		if source.tokenDisabled {
+			addAuth = " You can set GITHUB_TOKEN to a different token to make a request with a higher rate limit."
```
Comment (tgummerer) [inline sdk/go/common/workspace/plugins.go:477]:
```suggestion
			addAuth = " Your current GITHUB_TOKEN doesn't allow access to the repository, so we disabled it for this request.  You can set GITHUB_TOKEN to a different token to make a request with a higher rate limit."
```

Not sure if it's worth giving some more details.

Comment (Frassle) [reply sdk/go/common/workspace/plugins.go:477]: Why not

**Pull request merged**